base/workflow/R/runModule.run.write.configs.R

```diff
@@ -2,46 +2,127 @@
+      # generate new design
+      ensemble_size <- settings$ensemble$size %||% 1
       design_result <- PEcAn.uncertainty::generate_joint_ensemble_design(
         settings = settings[1],
         ensemble_size = ensemble_size
       )
-      input_design <- design_result$X
+      ens_input_design <- design_result$X
     }
+    
+    # handle SA design - check if already provided before generating
+    if (!is.null(sa_input_design)) {
+      # already provided, use as-is
+    } else if ("sensitivity.analysis" %in% names(settings)) {
+      # Load samples to determine SA run requirements
+      samples.file <- file.path(settings$outdir, "samples.Rdata")
+      load(samples.file)
+      
+      # Calculate total SA runs: 1 (median) + sum(quantiles per trait per pft)
+      num_sa_runs <- 1  # Start with median run
+      
+      for (pft_name in names(sa.samples)) {
+        if (pft_name == "env") next
+        
+        n_traits <- ncol(sa.samples[[pft_name]])
+        quantile_names <- rownames(sa.samples[[pft_name]])
+        n_quantiles <- sum(quantile_names != "50")  # Exclude median quantile
+        
+        num_sa_runs <- num_sa_runs + (n_traits * n_quantiles)
+      }
+
+      # Generate SA-specific input design
+      design_result_sa <- PEcAn.uncertainty::generate_joint_ensemble_design(
+        settings = settings[1],
+        ensemble_size = num_sa_runs
+      )
+      sa_input_design <- design_result_sa$X
+    }
+    
     return(PEcAn.settings::papply(settings,
                                   runModule.run.write.configs,
                                   overwrite = FALSE,
-                                  input_design = input_design))
+                                  input_design = NULL,
+                                  ens_input_design = ens_input_design,
+                                  sa_input_design = sa_input_design))
   } else if (PEcAn.settings::is.Settings(settings)) {
     # double check making sure we have method for parameter sampling
     if (is.null(settings$ensemble$samplingspace$parameters$method)) {
       settings$ensemble$samplingspace$parameters$method <- "uniform"
     }
-    if (is.null(input_design)) {
-      ensemble_size <- settings$ensemble$size
+    
+    # handle ensemble design - check in priority order
+    if (!is.null(ens_input_design)) {
+      # use provided design (from papply or direct call)
+    } else if (!is.null(input_design)) {
+      ens_input_design <- input_design
+    } else if ("ensemble" %in% names(settings)) {
+      # generate new design only if nothing provided
+      ensemble_size <- settings$ensemble$size %||% 1
       design_result <- PEcAn.uncertainty::generate_joint_ensemble_design(
         settings = settings,
         ensemble_size = ensemble_size
       )
-      input_design <- design_result$X
+      ens_input_design <- design_result$X
+    }
+    
+    # handle SA design - check if already provided before generating
```

Comment:
This is now turning into a fair amount of code repeated in both legs of the settings/multisettings if/else. Could we do some of this unconditionally outside the if? Or define a helper function to encapsulate it?